text/0000-introduce empty ranges.md

```diff
@@ -0,0 +1,267 @@
+- Feature Name: Introduction of empty Ranges
+- Start Date: 2022-05-29
+- RFC PR: (leave this empty)
+- Pony Issue: (leave this empty)
+
+# Summary
+
+The RFC carves out a subset of cases that are currently defined as *infinite* by proposing to instead produce an *empty* `Range` in these cases. This makes `Range` behave more similar to the way that Python's range() and Rust's Range() behave, allowing one to write more natural code. The proposal also addresses (fixes) a corner case that is currently unecessarily producing *infinite* Ranges.
```

Comment:
I'm not comfortable with the arguments in here that are "we are changing this because other languages do it this way".

Accepted RFCs set a precedent for making a change. "Other languages do it" isn't a good precedent.

I'd like to see things like "natural expectation" and "because language x does it" removed from the RFC and for the focus to be on Pony and why this change is good for Pony code and why those advantages are worth a breaking change.